kong/db/schema/others/declarative_config.lua

```diff
@@ -243,14 +243,12 @@ local function load_plugin_subschemas(fields, plugin_set, indent)
   for _, f in ipairs(fields) do
     local fname, fdata = next(f)
 
-    if fname == "plugins" then
+    -- Exclude cases where `plugins` are used expect from plugins entities.
+    -- This assumes other entities doesn't have `name` as its subschema_key.
+    if fname == "plugins" and fdata.elements and fdata.elements.subschema_key == "name" then
       for plugin in pairs(plugin_set) do
-        local _, err, err_t = plugin_loader.load_subschema(fdata.elements, plugin, errors)
+        local _, err = plugin_loader.load_subschema(fdata.elements, plugin, errors)
```

Comment:
Missing error handling?

Reply:
I think plugin_loader code only returns "ok, err" not third argument, so I removed them for clearity.

Reply:
oops, I missed the handling below 😆 apologies